backend/src/personnel/personnel.service.ts

```diff
@@ -221,53 +221,82 @@ export class PersonnelService {
 
     return availableDates;
   }
+
+  async getEventStartDate(personnelId: string, date: AvailabilityEntity): Promise<string> {
+    const start = await this.availabilityRepository.query('SELECT get_last_status_date_prior($1, $2, $3) as start_date', [personnelId, date.date, date.availabilityType]);
+    return format(start[0].start_date, 'yyyy-MM-dd');
+  }
+
+  async getEventEndDate(personnelId: string, date: AvailabilityEntity): Promise<string> {
+    const end = await this.availabilityRepository.query('SELECT get_last_status_date_after($1, $2, $3) as end_date', [personnelId, date.date, date.availabilityType]);
+    return format(end[0].end_date, 'yyyy-MM-dd');
+  }
+
   /**
    * Update the availability of a personnel for a specific date range for a specific avaiilability type
    * @param id
    * @param availability
    * @returns
    */
-  async updateAvailability(id: string, availability: UpdateAvailabilityDTO) {
-    const { from, to, type, deploymentCode } = availability;
+  async updateAvailability(id: string, availability: UpdateAvailabilityDTO):
+    Promise<{ updates: (UpdateResult | AvailabilityEntity)[], deleted?: DeleteResult }>
+  {
+    const { from, to, type, deploymentCode, removeFrom, removeTo } = availability;
 
     const startDate = parse(from, 'yyyy-MM-dd', new Date());
     const endDate = parse(to, 'yyyy-MM-dd', new Date());
 
-    const dates = [];
+
+    const getQb =
+      this.availabilityRepository.createQueryBuilder('availability')
+        .andWhere('date >= :startDate', { startDate: from })
```

Comment:
could do:

```
.where('date >= :startDate AND date <= :endDate', {startDate: from, endDate: to})
.andWhere('personnel = :id', {id})
```

To avoid chaining multiple ".andWhere's" (I think the first one should just be ".where" also)

Reply:
I won't be doing this!